expression/builtin_miscellaneous.go

```diff
@@ -529,28 +528,12 @@ func isIPv4(ip string) bool {
 	return true
 }
 
-type isIPv4CompatFunctionClass struct {
-	baseFunctionClass
-}
-
-func (c *isIPv4CompatFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinIsIPv4CompatSig{newBaseBuiltinFunc(args, ctx)}
-	return sig.setSelf(sig), errors.Trace(c.verifyArgs(args))
-}
-
-type builtinIsIPv4CompatSig struct {
-	baseBuiltinFunc
-}
-
-// eval evals a builtinIsIPv4CompatSig.
-// See https://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_is-ipv4-compat
-func (b *builtinIsIPv4CompatSig) eval(row []types.Datum) (d types.Datum, err error) {
-	args, err := b.evalArgs(row)
-	if err != nil {
-		return d, errors.Trace(err)
-	}
-
-	arg := args[0]
+// checkIPv6Prefix checks wheather the ipv6 values evaled by eval has special prefix
+// It will be used to check IPv4Mapped and IPv4Compat IPv6 Address
+// IPv4Mapped Address has prefix ::ffff:ipv4_address
+// Ipv4Compat Address has prefix ::ipv4_address
+func checkIPv6Prefix(args *[]types.Datum, v4InV6Prefix *[]byte) (d types.Datum, err error) {
```

Comment:
I think it may be better to use a `boolean argument` instead of `v4InV6Prefix`
to indicate `IPv4Mapped` or `IPv4Mapped`,
thus we can make the two v4InV6Prefix byte arrays be constants inside this function.